crates/spicepod/src/component/runtime.rs

```diff
@@ -102,14 +106,33 @@ pub struct TelemetryConfig {
 pub struct TaskHistory {
     #[serde(default = "default_true")]
     pub enabled: bool,
+    #[serde(default = "default_truncated")]
     pub captured_output: String,
+    #[serde(default = "default_retention_period")]
+    pub retention_period: String,
+    #[serde(default = "default_retention_check_interval")]
+    pub retention_check_interval: String,
+}
+
+fn default_truncated() -> String {
+    "truncated".to_string()
+}
+
+fn default_retention_period() -> String {
+    "1d".to_string()
+}
+
+fn default_retention_check_interval() -> String {
+    "30m".to_string()
 }
```

Comment:
nit: These can use an `Arc<str>` and we can just have a single static instance that gets reference cloned, instead of allocating each time these functions are called. see: https://github.com/spiceai/spiceai/blob/trunk/crates/db_connection_pool/src/dbconnection/snowflakeconn.rs#L72

Reply:
Or even better, could these be `&'static str`?